### PR TITLE
🧑‍💻 🔨 🔧 👷 Mise, Kind, and Tilt

### PR DESCRIPTION
**Pull request by rblaine95**
Well, this is a big one...

It's finally here, the migration away from `./manage up` and Docker Compose 🎉 

Standardize everything around [Mise](https://mise.jdx.dev/), [Kind](https://kind.sigs.k8s.io), and [Tilt](https://tilt.dev/), enabling Kubernetes Native Development with live reloading.

Still to discuss: Migrate Helm Charts from private repo to this repo.
This will make maintaining the Helm Charts _much_ easier.

Potential future improvements:
* Only run Regression Tests against EKS
* Run regular containerized tests against ephemeral Kind cluster

This will mean we can run multiple PRs at the same time, accelerating dev.

---

Mise, Kind, and Tilt

* Use Mise to install and manage tools
* Use Mise Tasks to install module dependencies
* Activate Mise managed Python venv
* Make Poetry version a build arg, disable pip caching

Docker Pull Through Caching

* Add Docker pull-through caching for:
  * docker.io
  * gcr.io
  * ghcr.io
  * quay.io
  * registry.k8s.io

* Delete docker cache container `kind:destroy`
* Only delete the actual cache on `kind:destroy-all`

Setup DBs, enable mise experimental

* Set mise experimental in `.mise.toml`
* Try cloning charts via SSH, fallback to HTTPS if failed
* Setup Redis, Postgres, and PGAdmin for CloudAPI

Setup Ledger

* Due to instability of running Ledger Nodes/Browser in Kind,
especially on resource restricted environments (github runners),
maintain using Docker Compose but only for Ledger Nodes/Browser

---

WRT Ledger Nodes/Browser:

I could add a check to see host resources (number of CPU cores) and
then decide whether or not to deploy Ledger Nodes/Browser in Kind.
Alternatively, I can add a CLI arg to decide between Compose Ledger
vs Kind Ledger.

### CODE REVIEW

File: dockerfiles/tests/Dockerfile [hunk 1]
```diff
@@ -1,7 +1,23 @@
-FROM python:3.12-slim
+FROM debian:12-slim
 
 WORKDIR /tests
 
+# Install Mise
+RUN apt-get update \
```
[the remark this answers is not in the document]
Comment (rblaine95) [reply dockerfiles/tests/Dockerfile:6]: That's the intent.

**Review by henrymsiska — APPROVED** [review]
Epic 🍓

File: docs/Quick Start Guide.md [hunk 2]
```diff
@@ -1,61 +1,60 @@
...
 
-This guide provides a simple walkthrough for starting, managing, and stopping a Docker-based project. Please ensure you have the necessary system requirements and a basic understanding of Docker and Docker Compose.
+This guide provides a simple walkthrough for starting, managing, and stopping a
+Docker-based project. Please ensure you have the necessary system requirements
+and a basic understanding of Docker and Docker Compose.
 
-1. Ensure you have Docker, Docker Compose, Bash, and (optionally) GNU Make installed on your machine. The project will require about 5GB of free disk space, 4GB of RAM, and an average CPU like an Intel i5.
+1. Ensure you have Docker, Docker Compose, Bash, and Mise installed on your
+machine. The project will require about 5GB of free disk space, 4GB of RAM, and
+an average CPU like an Intel i5.
```
Comment (cl0ete) [inline docs/Quick Start Guide.md:9]:
```suggestion
1. Ensure you have Docker, Docker Compose, Bash, and Mise installed on your
machine. The project will require about 25GB of free disk space, 16GB of RAM, and
an average CPU like an modern Intel i5 (min 4 cores).
```

File: README.md [hunk 3]
```diff
@@ -3,50 +3,163 @@
...
-./manage up
+mise run tilt:up
 ```
 
 Install the requirements:
 
 ```bash
-bash scripts/install_all_dependencies.sh
+mise run poetry:install:all
 ```
```
Comment (cl0ete) [inline README.md:164]: maybe add example.env with all the urls

**Review by cl0ete — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (rblaine95) [reply README.md:164]: @cl0ete, please check latest commits and let me know your thoughts

**Review by cl0ete — APPROVED** [review]

**Comment by wdbasson** [comment]
LGTM!

**Pull request merged**